### PR TITLE
src,test: add int64 test and fix JSRT

### PR DESCRIPTION
**Pull request by kfarnung**
Fixed a couple build warnings as well.

##### Checklist

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

**Comment by kfarnung** [comment]
CI: ~~https://ci.nodejs.org/job/chakracore-test-pull-request/218/~~
https://ci.nodejs.org/job/chakracore-test-pull-request/220/

File: src/node_api_jsrt.cc [hunk 1]
```diff
@@ -443,7 +419,7 @@ static napi_status SetErrorCode(JsValueRef error,
     CHECK_JSRT(JsHasProperty(error, namePropId, &hasProp));
 
     JsValueRef nameValue = JS_INVALID_REFERENCE;
-    std::array<JsValueRef, 2> args = { nameArray, JS_INVALID_REFERENCE };
+    std::array<JsValueRef, 2> args = {{ nameArray, JS_INVALID_REFERENCE }};
```
Comment (MSLaguana) [inline src/node_api_jsrt.cc]: Would you mind explaining this? was the initializer list previously hitting a different constructor?

**Review by boingoing — APPROVED** [review]
Looks alright. Plans to upsteam the unit tests? I assume they're compatible with v8.

**Review by MSLaguana — APPROVED** [review]

File: src/node_api_jsrt.cc (hunk 1 above)
Comment (kfarnung) [reply src/node_api_jsrt.cc]: clang (on macOS 10.13.4) specifically generated a warning that these braces should be doubled up.  Looking around it seems like most people suggest ignoring that warning and I see mention that clang 6.0 is specifically disabling the warning by default.

https://stackoverflow.com/questions/13905200/is-it-wise-to-ignore-gcc-clangs-wmissing-braces-warning

Given that I'll undo this change.

**Comment by kfarnung** [comment]
@boingoing I plan to open an upstream PR once this lands.

**Review by digitalinfinity — APPROVED** [review]

**Pull request merged**